database/test.go

```diff
@@ -84,7 +84,7 @@ func UpdateFlakyTests(db *sqlx.DB, commitID int) error {
 	stmt, err := db.Preparex(`
 	UPDATE tests SET flaky = true
 		WHERE tests.id in (
-			SELECT DISTINCT(tests.id), tests.name FROM results
+			SELECT DISTINCT(tests.id) FROM results
```

Comment:
Eamonn has investigated. 

- Your solution fixes the 2 column issue. 
- FYI: This snippet should have always thrown. Since noflake has worked at least once there must have been something to prevent calling this UPDATE, ie there should be some other code to SET flaky = true. But we did not find it immediately. In other words there might be something else fishy. But this will be object to another issue then.
- The GROUP BY statement is obsolete as we DISTINCT

Reply:
> The GROUP BY statement is obsolete as we DISTINCT

Isn't it the other way around? `DISTINCT(tests.id)` is redundant since we already GROUP BY (which is required to check if tests had successes AND failures) ?

> FYI: This snippet should have always thrown. Since noflake has worked at least once there must have been something to prevent calling this UPDATE, ie there should be some other code to SET flaky = true. But we did not find it immediately. In other words there might be something else fishy. But this will be object to another issue then.

Technically, it should have always thrown if there were tests which both failed AND succeeded for the same commitId. But why are flaky tests found if the update script does not work? 
Well, the flag `flaky` was added as a performance optimization (pre-calculating the flaky-ness). Test which were uploaded before that change got that flag from a migration script (`3_flaky_tests.up.sql`). The query in that file is correct.

Reply:
agree